go-controller/pkg/util/util.go

```diff
@@ -188,3 +198,30 @@ func HashForOVN(s string) string {
 	return fmt.Sprintf("a%s", hashString)
 
 }
+
+// UpdateIPsSlice will search for values of oldIPs in the slice "s" and update it with newIPs values of same IP family
+func UpdateIPsSlice(s, oldIPs, newIPs []string) []string {
+	n := make([]string, len(s))
+	copy(n, s)
+	for i, entry := range s {
+		for _, oldIP := range oldIPs {
+			if entry == oldIP {
+				for _, newIP := range newIPs {
+					if utilnet.IsIPv6(net.ParseIP(oldIP)) {
+						if utilnet.IsIPv6(net.ParseIP(newIP)) {
+							n[i] = newIP
+							break
```

Comment:
This break will have no effect because it will break the inner loop. I think here you want to put a label on line 206 and break there.
https://www.ardanlabs.com/blog/2013/11/label-breaks-in-go.html

Reply:
it does, because it breaks the inner loop, then the outer loop will break on line 222

Reply:
oh I missed the double break. Still, I think the labeled is more elegant solution (but not holding against it)

Reply:
yeah its not pretty, but we can clean it up later